<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>734</width>
    <height>482</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Учебный тренажер по алгоритмам сжатия данных</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>711</width>
      <height>421</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QComboBox" name="chooseAlg">
         <item>
          <property name="text">
           <string>Алгоритм LZW</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Алгоритм LZ78</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="1">
          <widget class="QRadioButton" name="chooseDec">
           <property name="text">
            <string>Режим декодрования</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">buttonGroup</string>
           </attribute>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QRadioButton" name="chooseEnc">
           <property name="text">
            <string>Режим кодирования</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">buttonGroup</string>
           </attribute>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Исходный текст</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextBrowser" name="input">
           <property name="readOnly">
            <bool>false</bool>
           </property>
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;abracadabra&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Результат</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextBrowser" name="output">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;justify&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Словарь</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextBrowser" name="dictionary">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;justify&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Описание шага</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextBrowser" name="description">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2"/>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="back">
         <property name="text">
          <string>Шаг назад</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pause">
         <property name="text">
          <string>Пауза</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="startButton">
         <property name="text">
          <string>Запуск</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="forvard">
         <property name="text">
          <string>Шаг вперед</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="begin">
         <property name="text">
          <string>Сначала</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <zorder>label_5</zorder>
   <zorder>verticalLayoutWidget_3</zorder>
   <zorder>verticalLayoutWidget_4</zorder>
   <zorder>begin</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>734</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="aboutAlg"/>
    <addaction name="aboutApp"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Самопроверка</string>
    </property>
    <addaction name="nextCode"/>
    <addaction name="nextDict"/>
   </widget>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>Сохранить как тест</string>
   </property>
  </action>
  <action name="action_3">
   <property name="text">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="aboutAlg">
   <property name="text">
    <string>Справка об алгоритме</string>
   </property>
  </action>
  <action name="aboutApp">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="nextCode">
   <property name="text">
    <string>Следующее кодовое слово</string>
   </property>
  </action>
  <action name="nextDict">
   <property name="text">
    <string>Словарь</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>chooseAlg</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeAlg(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseDec</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doChooseDec()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseEnc</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doChooseEnc()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>453</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextCode</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openQuestionNextCode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextDict</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openQuestionNextDict()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutApp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>begin</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doClear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>582</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>back</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doBackStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pause</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doPause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>380</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>forvard</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doNextStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>481</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>doStart()</slot>
  <slot>doNextStep()</slot>
  <slot>doClear()</slot>
  <slot>doChooseDec()</slot>
  <slot>doChooseEnc()</slot>
  <slot>doBackStep()</slot>
  <slot>openQuestionNextCode()</slot>
  <slot>openQuestionNextDict()</slot>
  <slot>changeAlg(int)</slot>
  <slot>doPause()</slot>
  <slot>showAbout()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
